vms/platformvm/block/executor/manager.go

```diff
@@ -103,3 +116,16 @@ func (m *manager) SetPreference(blockID ids.ID) (updated bool) {
 func (m *manager) Preferred() ids.ID {
 	return m.preferred
 }
+
+func (m *manager) VerifyTx(tx *txs.Tx) error {
```

Comment:
what I think it's a bit confusing here is that VerifyTx is to verify mempool transactions but we already have a MempoolTxVerifier visitor that basically does all of it already.
So maybe the change to be done is to move the "isBootstrapped" check into the mempoolVerified and that's it.
This would avoid increasing the blkManager interface (as per @danlaine 's comment) and make blkManager more of a backend object for verification of both mempool txs and txs in block.

Reply:
This function uses the `MempoolTxVerifier`. I don't think it's a good design pattern to required consumers of this package create a struct themselves to then call it. I think this pattern (https://github.com/ava-labs/avalanchego/blob/remove-add-unverified-tx/vms/platformvm/network/network.go#L162-L181) is much more digestible:

```golang
	// Verify the tx at the currently preferred state
	if err := n.manager.VerifyTx(tx); err != nil {
		n.ctx.Log.Debug("tx failed verification",
			zap.Stringer("txID", txID),
			zap.Error(err),
		)

		n.mempool.MarkDropped(txID, err)
		return err
	}
```

than the existing:
```golang
	verifier := txexecutor.MempoolTxVerifier{
		Backend:       b.txExecutorBackend,
		ParentID:      b.blkManager.Preferred(), // We want to build off of the preferred block
		StateVersions: b.blkManager,
		Tx:            tx,
	}
	if err := tx.Unsigned.Visit(&verifier); err != nil {
		b.MarkDropped(txID, err)
		return err
	}
```

Reply:
I don't really think it's more digestible but I'd suggest we find a proper name for `manager.VerifyTx` (my suggestion being `manager.VerifyMempoolCandidate`) and we defer discussions around mempool refactoring.